cli-script.php

```diff
@@ -1,152 +1,19 @@
-		}
-	}
-	$_REQUEST = array_merge($_REQUEST, $_GET);
-}
-
-// Set 'url' GET parameter
-if(isset($_SERVER['argv'][1])) {
-	$_REQUEST['url'] = $_SERVER['argv'][1];
-	$_GET['url'] = $_SERVER['argv'][1];
-}
-
-// require composers autoloader
-if (file_exists($autoloadPath = dirname(__DIR__) . '/vendor/autoload.php')) {
-	require_once $autoloadPath;
-}
-else  {
-	echo "Failed to include composer's autoloader, unable to continue\n";
-	exit(1);
-}
-
-/**
- * Include SilverStripe's core code
- */
-require_once("Core/Core.php");
-
-global $databaseConfig;
-
-// We don't have a session in cli-script, but this prevents errors
-$_SESSION = null;
-
-// Connect to database
-if(!isset($databaseConfig) || !isset($databaseConfig['database']) || !$databaseConfig['database']) {
-	echo "\nPlease configure your database connection details.  You can do this by creating a file
-called .env in " . BASE_PATH;
-	echo <<<ENVCONTENT
-
-Put the following content into this file:
---------------------------------------------------
-
-# Change this from 'dev' to 'live' for a production environment.
-SS_ENVIRONMENT_TYPE="dev"
-
-/* This defines a default database user */
-SS_DATABASE_SERVER="localhost"
-SS_DATABASE_USERNAME="<user>"
-SS_DATABASE_PASSWORD="<password>"
-SS_DATABASE_NAME="<database>"
---------------------------------------------------
-
-Once you have done that, run 'composer install' or './framework/sake dev/build' to create
-an empty database.
-
-For more information, please read this page in our docs:
-http://docs.silverstripe.org/en/getting_started/environment_management/
-
-
-ENVCONTENT;
-	exit(1);
-}
-DB::connect($databaseConfig);
-
-// register_argc_argv: Tells PHP whether to declare the argv & argc variables (that would contain the GET information)
-// http://php.net/manual/en/ini.core.php#ini.register-argc-argv
-if (!(int)ini_get('register_argc_argv')) {
-	echo 'register_argc_argv is required to obtain the GET information but is disabled.'
-	    .'Please enable it in your php.ini ('
-		. php_ini_loaded_file()
-        .')';
-    die();
-}
-
-// Get the request URL from the querystring arguments
-$url = isset($_SERVER['argv'][1]) ? $_SERVER['argv'][1] : null;
-if(!$url) {
-	echo 'Please specify an argument to cli-script.php/sake. For more information, visit'
-		. ' http://docs.silverstripe.org/en/developer_guides/cli' . "\n";
-	die();
-}
-
-$_SERVER['REQUEST_URI'] = BASE_URL . '/' . $url;
-
-// Direct away - this is the "main" function, that hands control to the apporopriate controller
-DataModel::set_inst(new DataModel());
-Director::direct($url, DataModel::inst());
-
-
+// CLI specific bootstrapping
+use SilverStripe\Core\AppKernel;
+use SilverStripe\Core\HTTPApplication;
+use SilverStripe\Core\Startup\OutputMiddleware;
+use SilverStripe\Control\HTTPRequest;
+
+require __DIR__ . '/src/includes/cli.php';
+require __DIR__ . '/src/includes/autoload.php';
+
+// Build request and detect flush
+$request = HTTPRequest::createFromEnvironment();
```

Comment:
Rather than using cli.php fake the environment and then calling this method, would it make more sense to have a separate class/method that will create a request from command-line args?